integration-cli/docker_cli_events_test.go

```diff
@@ -411,3 +412,247 @@ func checkEvents(t *testing.T, events []string) {
+//container := make(chan string)
+
+//wg.Add(1)
+//go func() {
+//defer wg.Done()
+
+//eventsCmd := exec.Command(dockerBinary, "events",
+//"-since", fmt.Sprint(startEpoch.Unix()))
+//stdout, err := eventsCmd.StdoutPipe()
+//err = eventsCmd.Start()
+//if err != nil {
+//t.Fatalf("failed to start 'docker events': %s", err)
+//}
+
+//go func() {
+//<-finish
+//eventsCmd.Process.Kill()
+//}()
+
+//goon <- true
+//containerID := <-container
+
+//var created, started, died, destroyed bool
+//matchCreate := regexp.MustCompile(containerID + "\\: \\(from busybox\\:latest\\) create$")
+//matchStart := regexp.MustCompile(containerID + "\\: \\(from busybox\\:latest\\) start$")
+//matchDie := regexp.MustCompile(containerID + "\\: \\(from busybox\\:latest\\) die$")
+//matchDestroy := regexp.MustCompile(containerID + "\\: \\(from busybox\\:latest\\) destroy$")
+
+//scanner := bufio.NewScanner(stdout)
+//for scanner.Scan() {
+//if ok := matchCreate.MatchString(scanner.Text()); ok {
+//if created {
+//t.Fatal("assertion fail: more than one container created")
+//}
+//close(eventCreate)
+//created = true
+//}
+//if ok := matchStart.MatchString(scanner.Text()); ok {
+//if started {
+//t.Fatal("assertion fail: more than one container started")
+//}
+//close(eventStart)
+//started = true
+//}
+//if ok := matchDie.MatchString(scanner.Text()); ok {
+//if died {
+//t.Fatal("assertion fail: more than one container died")
+//}
+//close(eventDie)
+//died = true
+//}
+//if ok := matchDestroy.MatchString(scanner.Text()); ok {
+//if destroyed {
+//t.Fatal("assertion fail: more than one container destroyed")
+//}
+//close(eventDestroy)
+//destroyed = true
+//}
+//}
+
+//err = eventsCmd.Wait()
+//if err != nil && !IsKilled(err) {
+//t.Fatalf("docker events had bad exit status: %s", err)
+//}
+//}()
+
+//select {
+//case <-time.After(30 * time.Second):
+//t.Fatal("waited too much!")
+//case <-goon:
+//// ignore, done
+//}
+
+//runCmd := exec.Command(dockerBinary, "run", "--rm", "busybox", "true")
+//out, _, err := runCommandWithOutput(runCmd)
+//if err != nil {
+//t.Fatal(out, err)
+//}
+//cleanedContainerID := stripTrailingCharacters(out)
+//container <- cleanedContainerID
+
+//select {
+//case <-time.After(30 * time.Second):
+//t.Fatal("failed to observe container create in timely fashion")
+//case <-eventCreate:
+//// ignore, done
+//}
+
+//select {
+//case <-time.After(30 * time.Second):
+//t.Fatal("failed to observe container start in timely fashion")
+//case <-eventStart:
+//// ignore, done
+//}
+
+//select {
+//case <-time.After(30 * time.Second):
+//t.Fatal("failed to observe container die in timely fashion")
+//case <-eventDie:
+//// ignore, done
+//}
+
+//select {
+//case <-time.After(30 * time.Second):
+//t.Fatal("failed to observe container destroy in timely fashion")
+//case <-eventDestroy:
+//// ignore, done
+//}
+
+//logDone("events - streamed to stdout")
+//}
+
+func TestEventsStreamingV2(t *testing.T) {
+	testRequires(t, SameHostDaemon)
```

Comment:
I think I'm almost done, pushed now, I'm now receiving 2 destroy event. I  think it's racing because of the `defer deleteAllContainers`.
Also, I'll clean that commented part!